prow/tide/tide.go

```diff
@@ -1122,9 +1189,10 @@ func (c *Controller) pickBatch(sp subpool, cc map[int]contextChecker, newBatchFu
 	// we must choose the oldest PRs for the batch
 	sort.Slice(sp.prs, func(i, j int) bool { return sp.prs[i].Number < sp.prs[j].Number })
 
-	var candidates []PullRequest
+	var candidates []CodeReviewCommon
 	for _, pr := range sp.prs {
-		if c.isRetestEligible(sp.log, &pr, cc[int(pr.Number)]) {
+		// isRetestEligible alters the input, so make a copy
+		if c.isRetestEligible(sp.log, pr.Clone(), cc[pr.Number]) {
```

Comment:
yes agreed. This is due to a "bug" that should cache prefetched HEAD OID but didn't. Since this PR is already huge I'm not sure that I should introduced the fix to this bug here. So addressing in https://github.com/kubernetes/test-infra/pull/26375

Reply:
Now that #26375 is merged, I'll rebase this PR and remove this `Clone` function